resources/prosody-plugins/token/util.lib.lua

```diff
@@ -458,6 +466,10 @@ function Util:verify_room(session, room_address)
             session.jitsi_meet_context_group,
             session.jitsi_web_query_prefix, session.jitsi_meet_str_tenant);
         session.jitsi_meet_tenant_mismatch = true;
+
+        if room_instance._data.allowUnauthenticatedAccess == false then
+            return false;
```

Comment:
Any reason not to return an error and a reason? I think we should also log it. Suggest moving this before the log above, so we log either "Token verification failed: tenant mismatch" or "Tenant differs..."

Reply:
yes as this should not show a login button.